### ISSUE TITLE
Spring formatting issue with list

### ISSUE THREAD

**Issue by marceloverdijk** [thymeleaf/thymeleaf#282]
I have a VatPercentage domain class containing:

```
@NotNull
@DecimalMin("0.00")
@DecimalMax("99.99")
@NumberFormat(style = Style.NUMBER, pattern = "#0.00")
private BigDecimal standardRate;
```

Note the @NumberFormat annotation to cause Spring to format the field nicely.
In my case with Dutch locale it will use a comma as decimal separator.

When using double-brakcets (as described in https://github.com/thymeleaf/thymeleaf/issues/223) it works when I have a single instance but not when looping over list.

To give example, in my controller I do this:

```
@RequestMapping(method = RequestMethod.GET)
public String list(Model model) {
    List<VatPercentage> vatPercentages = vatPercentageRepository.findAll(new Sort(DESC, "startDate"));
    model.addAttribute("vatPercentages", vatPercentages);
    model.addAttribute("vatPercentage", vatPercentages.get(0)); // just take first one
    return "admin/vatpercentages/list";
}
```

Now when I print this in my template like:

```
<p th:text="${{vatPercentage.standardRate}}"></p>

<ul th:each="vp : ${vatPercentages}">
    <li th:text="${{vp.standardRate}}"></li>
</ul>
```

I would expect that all fields printed will use a comma as decimal separator.

However only the first one (direct access, not via list) prints a comma. When looping over the list it prints a dot as decimal separator.

**Comment by danielfernandez** [comment]
Yes, this is in fact the expected behaviour in SpringMVC-based scenarios (works the same in JSP). This was already described at thymeleaf/thymeleaf-spring#37, and a specific action will be taken to solve this scenario as described in thymeleaf/thymeleaf-spring#49

Closing this one as duplicated, therefore.

**Comment by marceloverdijk** [comment]
Thx for quick reply. I wish it was easier on github to search issues...

I understand https://github.com/thymeleaf/thymeleaf-spring/issues/49 is not yet solved so I was trying to understand how the workaround this from issue #37.

I tried `th:text="${{vatPercentages[__${iteration.index}__].standardRate}}"` but it still does not use the formatting and still the dot is printed as decimal separator.

Maybe I'm misunderstanding the workaround...

**Comment by marceloverdijk** [comment]
Btw I think by using `<spring:eval />` in jsp the formatting also works within <c:forEach>.

How is thymeleaf outputting values when using double brackets `${{ }}` when using Spring? If it would go through a similar eval flow it should work.
Does Thymeleaf has a specific way to call the Spring eval mechanism?

**Comment by danielfernandez** [comment]
Actually, your comments made me look again at this problem, but this time from a different perspective. And this allowed me to solve the problem in an easy and very Spring-integrated way.

So this is already closed and this will be a part of 2.1.3. See thymeleaf/thymeleaf-spring#49 for more details.

2.1.3-SNAPSHOT has been updated, so you can use it for testing if you want, before 2.1.3 is published (will be published in a couple of days).

**Comment by marceloverdijk** [comment]
Sorry for late reply, but this now works 2.1.3. Thanks!
